Src/Particle/AMReX_ParticleTile.H

```diff
@@ -421,25 +421,25 @@ SoAParticle<NArrayReal, NArrayInt>::NextID ()
 #endif
     next = the_next_id++;
 
-    if (next > LastParticleID)
+    if (next > IntParticleIds::LastParticleID)
```

Comment:
If the user adds particles in large batches, then this check could miss an overflow after which next would be negative.

Reply:
Yes, fully agree. I am doubtful about a lot of these checks here.

The same problem is present for the (signed) `Long` ids.